hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaCache.java

```diff
@@ -436,6 +489,52 @@ private void updateQuotaFactors() {
         }
       }
     }
+
+    private void updateMachineQuotaFactors(int rsSize) {
+      if (rsSize != 0) {
+        // TODO if use rs group, the cluster limit should be shared by the rs group
+        machineQuotaFactor = 1.0 / rsSize;
+      }
+    }
+  }
+
+  static class RefreshableExpiringValueCache<T> {
+    private static final String DUMMY_KEY = "dummyKey";
```

Comment:
Can you explain what `DUMMY_KEY` is doing here? If I'm following, you have a `LoadingCache` (which essentially is a `Map` interface) that is only every populated with a single key, the value passed into the constructor as `String name`. The `CacheLoader` interface ignores the provided `key`, but, so what's with `DUMMY_KEY` ? Can't you use the provided `name` instead of `DUMMY_KEY` ? Or is there something else I'm missing here?

Reply:
The dummy key is just an ugly way to use guava caches as a more powerful supplier rather than a k,v function. I wrote that part before adding the logging & name field, so didn't think about this, but that's a nice idea to make this less ugly